python/tk_multi_launchapp/software_entity_launcher.py

```diff
@@ -25,6 +25,101 @@ def register_launch_commands(self):
+                    app_icon, self._tk_app
+                )
+                app_icon = sg_icon
+                self._tk_app.log_debug("App icon from ShotgunDataRetriever : %s" % app_icon)
+
+            app_engine = sw_entity["sg_engine"]
+            if app_engine:
+                # Try to retrieve the path to the specified engine. If nothing is
+                # returned, then this engine hasn't been loaded in the current
+                # environment, and there's not much more we can do.
+                app_engine_path = sgtk.platform.get_engine_path(
+                    app_engine, self._tk_app.sgtk, self._tk_app.context
+                )
+                if not app_engine_path:
+                    self._tk_app.log_warning(
+                        "Software engine %s is not loaded in the current environment. "
+                        "Cannot launch %s" % (app_engine, app_path)
+                    )
+                    continue
+
+            app_display_name = sw_entity["code"]
+            app_args = sw_entity[app_args_field] or ""
+            app_path = sw_entity[app_path_field]
+            register_cmds.extend(self._build_register_commands(
+                app_display_name, app_icon, app_engine, app_path, app_args, app_versions
+            ))
+
+        for rc in register_cmds:
+            self._register_launch_command(
+                rc["display_name"], rc["icon"], rc["engine"], rc["path"],
+                rc["args"], rc["version"]
+            )
+
+    def launch_from_path(self, path, version=None):
+        """
+        Entry point if you want to launch an app given a particular path.
+
+        :param path: File path DCC should open after launch.
+        :param version: (Optional) Specific version of DCC to launch.
+        """
+        # This functionality is not supported for Software entities.
+        self._tk_app.log_error(
+            "launch_from_path() is not supported by SoftwareEntityLauncher. "
+            "Please register individual application launch commands in your "
+            "Project's configuration to use this functionality."
+        )
+
+    def launch_from_path_and_context(self, path, context, version=None):
+        """
+        Entry point if you want to launch an app given a particular path
+        and context.
+
+        :param path: File path DCC should open after launch.
+        :param context: Specific context to launch DCC with.
+        :param version: (Optional) Specific version of DCC to launch.
+        """
+        # This functionality is not supported for Software entities.
+        self._tk_app.log_error(
+            "launch_from_path_and_context() is not supported by "
+            "SoftwareEntityLauncher. Please register individual application "
+            "launch commands in your Project's configuration to use this "
+            "functionality."
+        )
+
+
+    def _sg_software_entities(self):
```

Comment:
could we call this `_get_sg_software_entities()` instead? I am missing some sort of verb prefix for this method.